keynav.c

```diff
@@ -443,11 +466,25 @@ void parse_config() {
   char *homedir;
 
   keybindings = g_ptr_array_new();
+  startkeys = g_ptr_array_new();
   recordings = g_ptr_array_new();
 
   defaults();
   parse_config_file(GLOBAL_CONFIG_FILE);
   parse_config_file("~/.keynavrc");
+
+  // support XDG Base Directory
+  char *config_home = getenv("XDG_CONFIG_HOME");
+  char *user_config_file;
+
+  if (config_home &&
+      asprintf(&user_config_file, "%s/keynav/keynavrc", config_home) != -1) {
```

Comment:
asprintf is not available on Solaris (by default), but I suspect this is probably not a problem.

+1 to supporting XDG config pathing

Reply:
Cool!

Reply:
Thanks for the example of where asprintf will fail; I noted that it wasn't portable but I've only been testing on GNU and BSD systems. I've updated https://github.com/yjftsjthsd-g/keynav/issues/6 to note that it'll break on Solaris.